x/claims/keeper/claim.go

```diff
@@ -99,30 +99,23 @@ func (k Keeper) MergeClaimsRecords(
 	// the recipient or sender has completed an action but the other hasn't
 	actions := []types.Action{types.ActionVote, types.ActionDelegate, types.ActionEVM, types.ActionIBCTransfer}
 	for _, action := range actions {
-		senderCompleted := senderClaimsRecord.HasClaimedAction(action)
-		recipientCompleted := recipientClaimsRecord.HasClaimedAction(action)
 
-		switch {
-		case senderCompleted && recipientCompleted:
-			// Both sender and recipient completed the action.
-			// Only mark the action as completed
-			mergedRecord.MarkClaimed(action)
+		// Safety check: the sender record cannot have any claimed actions, as
+		//  - the sender is not an evmos address and can't claim vote, delegation or evm actions
+		//  - the first attempt to perform an ibc callback from the senders account will merge/migrate the entire claims record
+		if senderCompleted := senderClaimsRecord.HasClaimedAction(action); senderCompleted {
+			return types.ClaimsRecord{}, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "non-evmos sender must not have claimed action: %v", action)
```

Comment:
can we change the error type `sdkerrors.ErrUnknownRequest` for something more descriptive?